pkg/targets/adapter/azuresentineltarget/adapter.go

```diff
@@ -0,0 +1,150 @@
+)
+
+type azuresentineltargetAdapter struct {
+	client         *http.Client
+	clientID       string
+	tenantID       string
+	azureCreds     string
+	subscriptionID string
+	resourceGroup  string
+	workspace      string
+	clientSecret   string
+
+	// sink     string
+	replier  *targetce.Replier
+	ceClient cloudevents.Client
+	logger   *zap.SugaredLogger
+}
+
+// EnvAccessorCtor for configuration parameters
+func EnvAccessorCtor() pkgadapter.EnvConfigAccessor {
+	return &envAccessor{}
+}
+
+// NewTarget adapter implementation
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*envAccessor)
+	logger := logging.FromContext(ctx)
+
+	replier, err := targetce.New(env.Component, logger.Named("replier"),
+		targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+		targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeAzureSentinelTargetGenericResponse),
+		targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(env.CloudEventPayloadPolicy)))
+	if err != nil {
+		logger.Panicf("Error creating CloudEvents replier: %v", err)
+	}
+
+	return &azuresentineltargetAdapter{
+		client:         http.DefaultClient,
+		clientID:       env.ClientID,
+		tenantID:       env.TenantID,
+		azureCreds:     env.ClientSecret,
+		subscriptionID: env.SubscriptionID,
+		resourceGroup:  env.ResourceGroup,
+		workspace:      env.Workspace,
+		clientSecret:   env.ClientSecret,
+
+		// sink:     env.Sink,
+		replier:  replier,
+		ceClient: ceClient,
+		logger:   logger,
+	}
+}
+
+var _ pkgadapter.Adapter = (*azuresentineltargetAdapter)(nil)
+
+// Returns if stopCh is closed or Send() returns an error.
+func (a *azuresentineltargetAdapter) Start(ctx context.Context) error {
+	a.logger.Info("Starting AzureSentinel Target Adapter")
+	return a.ceClient.StartReceiver(ctx, a.dispatch)
+}
+
+func (a *azuresentineltargetAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
+	typ := event.Type()
+	if typ != v1alpha1.EventTypeAzureSentinelTargetIncident {
+		return a.replier.Error(&event, targetce.ErrorCodeEventContext, fmt.Errorf("event type %q is not supported", typ), nil)
+	}
+
+	i := &Incident{}
+	if err := event.DataAs(i); err != nil {
+		a.logger.Errorf("decoding event: %v", err)
+		return nil, nil
+	}
+
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		a.logger.Errorf("creating Azure authorizer: %v", err)
+		return nil, nil
+	}
+
+	reqBody, err := json.Marshal(*i)
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "marshaling request for retrieving an access token")
+	}
+
+	rURL := `https://management.azure.com/subscriptions/` + a.subscriptionID + `/resourceGroups/` + a.resourceGroup + `/providers/Microsoft.OperationalInsights/workspaces/` + a.workspace + `/providers/Microsoft.SecurityInsights/incidents/` + uuid.New().String() + `?api-version=2020-01-01`
```

Comment:
Whoa, this is an ugly one! 😨 
How about replacing it with something more multilinear? 

```suggestion
	rURL := fmt.Sprintf("https://management.azure.com/"+
		"subscriptions/%s/resourceGroups/%s/"+
		"providers/Microsoft.OperationalInsights/workspaces/%s/"+
		"providers/Microsoft.SecurityInsights/incidents/%s?api-version=2020-01-01",
		a.subscriptionID, a.resourceGroup, a.workspace, uuid.New().String())
```
Feel free to play with the line lengths.